.evergreen/check-all.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```

Comment:
the configure-rust script uses bash-only syntax; I took the opportunity to standardize all of our scripts to use bash (even if they don't call configure-rust.sh) to try to avoid any problems around this in the future